packages/imperative/src/rest/src/session/ConnectionPropsForSessCfg.ts

```diff
@@ -139,12 +139,16 @@ export class ConnectionPropsForSessCfg {
             promptForValues.push("hostname");
         }
 
-        if (ConnectionPropsForSessCfg.propHasValue(sessCfgToUse.port) === false && !doNotPromptForValues.includes("port")) {
+        if ((ConnectionPropsForSessCfg.propHasValue(sessCfgToUse.port) === false || sessCfgToUse.port === 0) &&
```

Comment:
The change to add `sessCfgToUse.port === 0` was part of [a separate PR](https://github.com/zowe/zowe-cli/pull/2077) that's unrelated to the SSH one. It was documented as a breaking change, so I don't think we want to port it to v2.